src/apps/dashboard/modal/ready-for-loan-modal/ready-to-loan-modal-content.tsx

```diff
@@ -0,0 +1,237 @@
+        setPhysicalReservationsReadyToLoan(readyToLoan);
+      }
+    }
+  }, [physicalReservations, physicalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (digitalReservations && !digitalReservationsReadyToLoan) {
+      const { reservations } = digitalReservations;
+      if (reservations) {
+        const readyToLoan = reservations.filter((reservation) => {
+          const expiryDate = dayjs(reservation.expireDateUtc);
+          return expiryDate > today;
+        });
+        if (readyToLoan) {
+          setDigitalReservationsReadyToLoan(readyToLoan);
+        }
+      }
+    }
+  }, [digitalReservations, digitalReservationsReadyToLoan, today]);
+
+  useEffect(() => {
+    if (
+      physicalReservationsReadyToLoan &&
+      digitalReservationsReadyToLoan &&
+      !allSelectableReservations
+    ) {
+      const fausts = physicalReservationsReadyToLoan.reduce((acc, pr) => {
+        return { ...acc, [pr.recordId]: pr.reservationId };
+      }, {});
+      const identsObj = digitalReservationsReadyToLoan.reduce((acc, dr) => {
+        return { ...acc, [dr.identifier as string]: dr.identifier };
+      }, {});
+      const selectableReservations = { ...fausts, ...identsObj };
+      if (Object.keys(selectableReservations).length > 0) {
+        setAllSelectableReservations(
+          selectableReservations as {
+            [key: string]: string;
+          }[]
+        );
+      }
+    }
+  }, [
+    allSelectableReservations,
+    physicalReservationsReadyToLoan,
+    digitalReservationsReadyToLoan
+  ]);
+
+  const selectAllQueuedResevationsHandler = () => {
+    if (Object.keys(selectedReservations).length > 0) {
+      setSelectedReservations([]);
+    } else if (allSelectableReservations) {
+      setSelectedReservations(allSelectableReservations);
+    }
+  };
+  const setCustomSelection = useCallback(
+    (elementId: string, reservationId: string) => {
+      if (Object.keys(selectedReservations).includes(elementId as string)) {
+        const newSelection = { ...selectedReservations };
+        delete newSelection[elementId as never];
+        setSelectedReservations(newSelection);
+      } else {
+        const updatedSelectedReservations = {
+          ...selectedReservations,
+          ...{
+            [elementId]: reservationId
+          }
+        };
+        setSelectedReservations(updatedSelectedReservations);
+      }
+    },
+    [selectedReservations]
+  );
+  const removeSelectedReservations = () => {
+    const selectedReservationsKeys = Object.keys(selectedReservations);
+    const selectedReservationsValues = Object.values(selectedReservations);
+    if (selectedReservationsKeys.length > 0) {
+      selectedReservationsKeys.map((reservation) => {
+        const index = selectedReservationsKeys.indexOf(reservation);
+        const reservationToDelete = selectedReservationsValues[index];
+        switch (reservation.length) {
+          case 8: // Physical Loan on faust
```

Comment:
We should have a configuration for these reservation lengths. It doesnt make sense that business logic like this is "randomly" coded